app/routes/questionnaire.py

```diff
@@ -179,14 +182,47 @@ def get_section(schema, questionnaire_store, section_id, list_item_id=None):
         questionnaire_store.metadata,
     )
 
-    if not schema.is_hub_enabled():
-        redirect_location = router.get_first_incomplete_location_in_survey()
-        return redirect(redirect_location.url())
+    if request.method == "POST":
+        if schema.is_hub_enabled():
+            return redirect(url_for(".get_questionnaire"))
+        return redirect(router.get_first_incomplete_location_in_survey().url())
 
     if section_id not in router.enabled_section_ids:
         return redirect(url_for(".get_questionnaire"))
 
     routing_path = router.routing_path(section_id=section_id, list_item_id=list_item_id)
+
+    if router.can_access_section_summary(section_id, list_item_id):
+        list_name = schema.get_repeating_list_for_section(section_id)
+        section_title = schema.get_title_for_section(section_id)
+        location = Location(
+            section_id=section_id, list_name=list_name, list_item_id=list_item_id
+        )
+        section_summary_context = SectionSummaryContext(
+            flask_babel.get_locale().language,
+            schema,
+            questionnaire_store.answer_store,
+            questionnaire_store.list_store,
+            questionnaire_store.progress_store,
+            questionnaire_store.metadata,
+        )
+        context = section_summary_context(location)
+
+        return _render_page(
+            block_type="SectionSummary",
+            context=context,
+            current_location=location,
+            previous_location_url=router.get_section_return_location_when_section_complete(
+                routing_path
+            ).url(),
+            schema=schema,
+            page_title=section_title,
+        )
+
+    if not schema.is_hub_enabled():
+        redirect_location = router.get_first_incomplete_location_in_survey()
+        return redirect(redirect_location.url())
```

Comment:
As mentioned earlier, we should consider having a handler that encapsulates a lot of the logic in this method.